pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -285,20 +286,21 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 		return r.getRequeueResult(externalSecret), nil
 	}
 
-	// update status of the ExternalSecret when this function returns, if needed.
+	// patch status of the ExternalSecret when this function returns, if needed.
 	// NOTE: we use the ability of deferred functions to update named return values `result` and `err`
 	// NOTE: we dereference the DeepCopy of the status field because status fields are NOT pointers,
 	//       so otherwise the `equality.Semantic.DeepEqual` will always return false.
-	currentStatus := *externalSecret.Status.DeepCopy()
+	prevExternalSecret := *externalSecret.DeepCopy()
+
 	defer func() {
 		// if the status has not changed, we don't need to update it
-		if equality.Semantic.DeepEqual(currentStatus, externalSecret.Status) {
+		if equality.Semantic.DeepEqual(prevExternalSecret.Status, externalSecret.Status) {
 			return
 		}
 
-		// update the status of the ExternalSecret, storing any error in a new variable
+		// patch the status of the ExternalSecret, storing any error in a new variable
 		// if there was no new error, we don't need to change the `result` or `err` values
-		updateErr := r.Status().Update(ctx, externalSecret)
+		updateErr := r.Status().Patch(ctx, externalSecret, client.MergeFrom(&prevExternalSecret))
```

Comment:
Please do not use patch...

There is a reason we use Update only, and never Patch (unless we cant avoid it because we dont know the full object).

The reason is that we want to get a conflict and fail to update if we have an outdated copy of the resource, a patch will always succeed so might remove state from the status.